### ISSUE TITLE
simple-salesforce report import error

### ISSUE THREAD

**Issue by ycicek-firefly** [simple-salesforce/simple-salesforce#584]
Hi,

I have been using the code below to import my reports and manipulate them as a pandas dataframe. 

`#Salesforce Connection
sf = Salesforce(username = '------',password= '-------' , security_token= '--------')

sf_instance = '------' #Your Salesforce Instance URL
reportId = '------' # add report id
export = '?isdtp=p1&export=1&enc=UTF-8&xf=csv'
sfUrl = sf_instance + reportId + export
session = requests.Session()
session.max_redirects = 1000
response = session.get(sfUrl, headers=sf.headers, cookies={'sid': sf.session_id})
download_report = response.content.decode('utf-8')
sf_data = pd.read_csv(StringIO(download_report))
sf_data.head(n=1000)`

For the last 2 days, I have been encountering the following error:
'TooManyRedirects: Exceeded 30 redirects.'

I have also tried to increase max_redirects cap to 1000, still the same outcome unfortunately.

I haven't edited the code, my credentials are unchanged. What might be the cause of the problem, is anyone else experiencing this?

**Comment by hemenkapadia** [comment]
I am experiencing the same behavior @ycicek-firefly. Which version of simple_salesforce are you using ?

I am using 1.11.2 on python 3.8.13

**Comment by ycicek-firefly** [comment]
I am using version 1.11.6 on Python 3.9.7

**Comment by kami-artik** [comment]
I am having the same issue starting from Saturday. I think it is related to the Salesforce update.

**Comment by stephtuo** [comment]
Same issue here

**Comment by kami-artik** [comment]
Does anyone know if there is any other way to export SF reports with python?
Thanks!

**Comment by hkajiwara** [comment]
I did quick tests on  both Summer'22 and Winter'23 orgs, thus the export seems to work. Salesforce.sf_instance should return My Domain and it would work. To narrow down the issue, does putting the same export URL to your browser work?

```
sf = Salesforce(username=<username>, password=<password>, security_token='')
export_url = 'https://' + sf.sf_instance + '/' + '<report id>?isdtp=p1&export=1&enc=UTF-8&xf=csv'

session = requests.Session()
response = session.get(export_url, 
                       headers=sf.headers, 
                       cookies={'sid': sf.session_id})
download_report = response.content.decode('utf-8')
print(download_report)
```

**Comment by ycicek-firefly** [comment]
@hkajiwara It worked my friend, kudos to you!

**Comment by kami-artik** [comment]
I saw an another solution under this 
https://stackoverflow.com/questions/74087168/how-to-solve-python-code-error-toomanyredirects-exceeded-30-redirects

I have changed my sf_org from <your_enterprise>.lightning.force.com to <your_enterprise>.my.salesforce.com

**Comment by TobiasCortese** [comment]
I've made the updates indicated above but I'm getting the following error. 
Has anyone encountered the same?
Perhaps there's a lever that needs to be pulled within Salesforce in order to enable connecting via the new enhanced domain?

`Failed to establish a new connection: [Errno -2] Name or service not known`

**Comment by JeffB-Mountaineers** [comment]
@TobiasCortese Could it be an MFA requirement? Do we need to generate a security token to use?

**Comment by jon-wobken** [comment]
@TobiasCortese - can you confirm if Salesforce is return a session_id?

In addition, did your export url use the format below? The sf_instance does not return a https or have a "/" at the end

export_url = 'https://' + sf.sf_instance + '/' + '<report id>?isdtp=p1&export=1&enc=UTF-8&xf=csv'

**Comment by TobiasCortese** [comment]
@jon-wobken I'm embarrassed to admit that I failed to double-check the export_url and thus notice that I neglected to add the closing "/" :-(

Thanks for your help, patience and understanding!

**Comment by krzcho** [comment]
Now I am getting just the header, without any rows; any ideas?

**Comment by wdaher** [comment]
> Now I am getting just the header, without any rows; any ideas?

Do you have computed columns/a formula in your sheet? I found that the CSV export didn't end up working when I had a formula column.

**Comment by krzcho** [comment]
Yes, there is grouping. Export to CSV from the browser itself works fine. Just not from the library. Browser uses POST instead of GET - not sure if it makes the diff...
